### PR TITLE
GA Integration: Support Event Dimensions and Metrics

### PR DESCRIPTION
**Pull request by rjhilgefort**
Google Analytics supports tracking custom dimensions and metrics through page views as well as events. The integration currently allows this through `trackPage` because there's not minipulation of the keys passed in. `trackEvent` modifies keys by prepending `event` (per spec)- it does not, however, look for the presence of the special `dimension` and `metric` keys. This PR adds support by not modifying those keys when found.

Here's a link to the GA spec on how they should be passed: https://developers.google.com/analytics/devguides/collection/analyticsjs/custom-dims-mets

### CODE REVIEW

**Comment by rjhilgefort** [comment]
The test appears to be failing on `npm install` and is unrelated to the changes I'm making in this PR.

File: addon/integrations/google-analytics.js [hunk 1]
```diff
@@ -59,8 +59,13 @@ export default Base.extend({
     }
 
     for (let key in options) {
-      const capitalizedKey = capitalize(key);
-      gaEvent[`event${capitalizedKey}`] = options[key];
+      const value = options[key];
+
+      // If key is not a 'dimension' or 'metric', prepend with 'event'
+      let shouldPrefix = !/^(dimension|metric)[0-9]{1,2}/.test(key);
```
Comment (rjhilgefort) [inline addon/integrations/google-analytics.js]: While the variable is not strictly necessary, I believe it improves readability (as opposed to having the test directly in the `if`` statement).

**Comment by tomasbasham** [comment]
I guess there is an issue with travis. I have come across this before and it just sorted itself out eventually. I'll leave it another day before re-trying the test. However running tests locally shows an error:

```
not ok 5 PhantomJS 2.1 - JSHint - modules/ember-cli-analytics/integrations/google-analytics.js: should pass jshint
    ---
        actual: >
            false
        expected: >
            true
        message: >
            modules/ember-cli-analytics/integrations/google-analytics.js should pass jshint.
            modules/ember-cli-analytics/integrations/google-analytics.js: line 66, col 25, Expected '{' and instead saw 'key'.

            1 error
        Log: |
    ...
```

File: addon/integrations/google-analytics.js (hunk 1 above)
Comment (tomasbasham) [reply addon/integrations/google-analytics.js]: I agree

Comment (tomasbasham) [reply addon/integrations/google-analytics.js]: Although I would use `const` over `let` simply due to the fact we know it wont be changed during the runtime and, of course, neither do we want it to be

Comment (rjhilgefort) [reply addon/integrations/google-analytics.js]: 👍

**Comment by rjhilgefort** [comment]
Updated based on feedback. All tests are passing locally. Fingers crossed Travis agrees.

**Comment by tomasbasham** [comment]
Great, man. Travis still being a nuisance. I'll jump on their support tomorrow if it is still failing. But I can see it all works 😀

**Comment by tomasbasham** [comment]
There we go. Travis has stopped throwing it's toys out of the pram. I'll merge this in ASAP

**Comment by rjhilgefort** [comment]
👍 and thanks for the repo!

**Comment by rjhilgefort** [comment]
@tomasbasham Don't plan on merging?

**Comment by tomasbasham** [comment]
I have merged it

**Comment by rjhilgefort** [comment]
👍

**Pull request closed without merging**